packages/SwingSet/src/vats/timer/types.js

```diff
@@ -1,198 +1 @@
- * matches and return a RelativeTime labeled with that brand.
- *
- * @property {(abs: Timestamp, rel: RelativeTime) => Timestamp} addAbsRel
- * An absolute time + a relative time gives a new absolute time.
- *
- * @property {(rel1: RelativeTime, rel2: RelativeTime) => RelativeTime} addRelRel
- * A relative time (i.e., a duration) + another relative time
- * gives a new relative time.
- *
- * @property {(abs1: Timestamp, abs2: Timestamp) => RelativeTime} subtractAbsAbs
- * The difference between two absolute times is a relative time. If abs1 > abs2
- * the difference would be negative, so this method throws instead.
- *
- * @property {(abs1: Timestamp, abs2: Timestamp) => RelativeTime} clampedSubtractAbsAbs
- * The difference between two absolute times is a relative time. If abs1 > abs2
- * the difference would be negative, so this method returns a zero
- * relative time instead.
- *
- * @property {(abs: Timestamp, rel: RelativeTime) => Timestamp} subtractAbsRel
- * An absolute time - a relative time gives a new absolute time
- *
- * @property {(rel1: RelativeTime, rel2: RelativeTime) => RelativeTime} subtractRelRel
- * The difference between two relative times.
- *
- * @property {(rel: RelativeTime) => boolean} isRelZero
- * Does it represent a zero relative time, i.e., the difference
- * of an absolute time with itself? (We choose not to define a similar
- * isAbsZero, even though we could, because it is much less likely to be
- * meaningful.)
- *
- * @property {(rel: RelativeTime, nat: bigint) => RelativeTime} multiplyRelNat
- * @property {(rel: RelativeTime, nat: bigint) => RelativeTime} divideRelNat
- * @property {(rel1: RelativeTime, rel2: RelativeTime) => bigint} divideRelRel
- *
- * @property {(abs: Timestamp, step: RelativeTime) => RelativeTime} modAbsRel
- * An absolute time modulo a relative time is a relative time. For example,
- * 20:17 on July 20, 1969 modulo 1 day is just 20:17, a relative time that
- * can be added to the beginning of any day.
- *
- * @property {(rel: RelativeTime, step: RelativeTime) => RelativeTime} modRelRel
- * A relative time modulo a relative time is a relative time. For example,
- * 3.5 hours modulo an hour is 30 minutes.
- *
- * @property {(abs1: Timestamp, abs2: Timestamp) => RankComparison} compareAbs
- * Compares two absolute times. This comparison function is compatible
- * with JavaScript's `Array.prototype.sort` and so can be used to sort an
- * array of absolute times. The result is -1, 0, or 1 indicating whether
- * the first argument is less than, equal, or greater than the second.
- *
- * @property {(rel1: RelativeTime, rel2: RelativeTime) => RankComparison} compareRel
- * Compares two relative times. This comparison function is compatible
- * with JavaScript's `Array.prototype.sort` and so can be used to sort an
- * array of relative times. The result is -1, 0, or 1 indicating whether
- * the first argument is less than, equal, or greater than the second.
- */
+// JS module for types.d.ts
```

Comment:
Don't we need a single `export {}` in this file to be backwards compatible?

Reply:
If the file is never actually imported, let's drop it completely?

Reply:
Please remove this file if unused (not in any other `import './types.js'`) or add an `export {};` as content.

Reply:
Without this the types aren't available outside the module.

Reply:
I do not understand how the presence of this file helps for packages outside `SwingSet`. This feels magical to me. And I'm concerned about the file not respecting the module semantics (presence of an `import` or `export` statement) since the `.d.ts` has module semantics.